DataFormats/Detectors/TPC/include/DataFormatsTPC/WorkflowHelper.h

```diff
@@ -62,6 +62,21 @@ struct getWorkflowTPCInput_ret {
 };
 } // namespace internal
 
+static auto getWorkflowTPCInputEmpryPtr()
+{
+  /*
+  In case the need of TPCInput is conditional, this will allow to define an empty unique_ptr to this complex structure outside 
+  of the "if" scope, i.e. 
+  auto inp = getWorkflowTPCInputEmpryPtr();
+  if (want_TPCInput) {
+    inp = getWorkflowTPCInput(...);
+    consumer->setInp(inp);
+  }
+*/
+  std::unique_ptr<internal::getWorkflowTPCInput_ret> ptr;
+  return ptr;
+}
+
```

Comment:
Here is a typo `Emry`.
And can't you just use decltype:
```
std::decay_t<decltype(o2::tpc::getWorkflowTPCInput(pc))> optPtr;
if (cont) {
  optPtr = o2::tpc::getWorkflowTPCInput(...)
}
...
```

Reply:
@davidrohr thanks for the hint, indeed, works with ``std::decay``. I was trying with ``std::result_of<decltype(o2::tpc::getWorkflowTPCInput(pc))>::type`` but this did not work...